### PR TITLE
reintroduce pollwatcher now that rust 1.1 is out

### LINKED ISSUE #12: Remake PollWatcher
For those not familiar with the background who want to help:

What was done:

- Support was achieved for both Linux and OS X.
- Windows native support (#4) may be worked on.
- The polling backend requires the ability to get the `mtime` of a file, which cannot currently be done in stable Rust, so it was dropped.

What is needed:

- To get a replacement for `stat.modified()`.
- To implement anew (or adapt from the old code) a polling backend.

### PR DESCRIPTION
**Pull request by blaenk**
Closes #12

This uses @acrichto's `filetime` crate for modified time fetching.

### CODE REVIEW

**Comment by blaenk** [comment]
I'm not sure what @maurizi's status is on the windows watcher, that would be the preferred way, but I think a poll watcher is still good to have, if not just in the meantime, then as a reference implementation of sorts.

**Comment by maurizi** [comment]
I'm working on the windows watcher, but it still has some kinks in it and life is interfering with me getting around to finishing it at the moment.  I should be able to get back to it soon though.

A polling solution is a good idea anyway, for more esoteric platforms like BSD, or for cases where the the OS file watching APIs do not work properly, like when the watched directory is on a VirtualBox shared folder.

**Comment by blaenk** [comment]
Awesome, I'm rooting for you @maurizi :+1:

File: README.md [hunk 1]
```diff
@@ -50,8 +50,8 @@ fn main() {
 ## Platforms
 
 - Linux / Android: inotify
-- ~~All platforms: polling~~ (not working, see notice)
 - OS X: FSEvent
+- All platforms: polling
```
Comment (passcod) [inline README.md]: Can you also remove the notice at the top while we're at it?

Comment (blaenk) [inline README.md]: Will do when I get back.

On Thursday, June 25, 2015, Félix Saparelli <notifications@github.com>
wrote:

> In README.md
> <https://github.com/passcod/rsnotify/pull/19#discussion_r33318762>:
>
> >  - OS X: FSEvent
> > +- All platforms: polling
>
> Can you also remove the notice at the top while we're at it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/passcod/rsnotify/pull/19/files#r33318762>.
>


-- 
- Jorge Israel Peña

**Comment by passcod** [comment]
Great job! Thank you so much.

I'll open a ticket for the various TODOs in the code.

**Comment by blaenk** [comment]
Alright I removed the notice.

**Comment by passcod** [comment]
Awesome!

**Comment by passcod** [comment]
I'll do a release later today with #14 and #18 also merged in.

**Pull request merged**